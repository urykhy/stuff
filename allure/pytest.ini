[pytest]
markers =
    minor: test allure marks
