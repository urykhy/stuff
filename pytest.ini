[pytest]
log_cli = 1
log_cli_level = 30
markers =
    slow: mark test as slow.
