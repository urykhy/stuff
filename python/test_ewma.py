import pytest

import os
import sys

from pathlib import Path

sys.path.append(os.path.join(Path(__file__).resolve().parent, "build"))

from libewma_py import Ewma


def test_ewma(snapshot):
    slow = Ewma(0.95, 0.1)
    fast = Ewma(0.95 * 0.75, 0.1)

    for i in range(0, 5):
        slow.add(0.2)  # latency 0.2 now
        fast.add(0.2)
    assert slow.estimate() == snapshot
    assert fast.estimate() == snapshot
    assert slow.estimate() < fast.estimate()

    for i in range(5, 10):
        slow.add(0.1)  # latency 0.1 again
        fast.add(0.1)
    assert slow.estimate() == snapshot
    assert fast.estimate() == snapshot
