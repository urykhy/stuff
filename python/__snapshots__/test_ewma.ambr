# serializer version: 1
# name: test_ewma
  0.12262190625000002
# ---
# name: test_ewma.1
  0.18163781564331058
# ---
# name: test_ewma.2
  0.11750439982616213
# ---
# name: test_ewma.3
  0.11499048621319893
# ---
